### PR TITLE
feat: day 1 logique

### PR DESCRIPTION
**Pull request by GabinDDL**
# feat : day 1 logique

## Description

Course 1 from the beginning of the axioms of orders to the last written examples of good orders have been written

## Changes Made
Add theoremstyle lemma and add definition, proposition, lemma and example to logic order course 1

## Checklist

- [x] I have self-reviewed my code
- [x] Code follows project's style guidelines
- [x] Tests added and passing
- [x] Documentation updated (if needed)

## @Mention 

@Yag000

### CODE REVIEW

File: logique-notes.tex [hunk 1]
```diff
@@ -43,4 +45,146 @@
...
+    \end{itemize}
+    On peut aussi l'axiomatiser de la manière suivante :
+
+    $<$ est une relation d'ordre strict sur $E$ si elle est :
+    \begin{itemize}
+        \item Irréflexive : $x \nless x$
+        \item Transitive : $\forall x$, $y$, $z \in E$, ($x < y$ et $y < z$) $\implies x < z$
+
+    \end{itemize}
+
```
Comment (Yag000) [inline logique-notes.tex:78]: A nice remark is missing, the one that defines $\leq$ based on the $<$ axioms:

```latex
$$ x \leq y \text{ssi} x < y \text{or} x = y $$
```

File: logique-notes.tex [hunk 2]
```diff
@@ -43,4 +45,146 @@
...
+\begin{definition}[Relation totale]
+
+    On dit qu'une relation d'ordre $\leqslant$ est totale sur $E$ si pour $\forall x$, $y \in E$, x et y sont comparables.
+
+\end{definition}
+
+\begin{lemma}
+
+    Soit $<$ une relation d'ordre strict, soit $\leqslant$ l'ordre large associé. Donc $\leqslant$ est totale si et seulement si :
```
Comment (Yag000) [inline logique-notes.tex]:
```suggestion
    Soit $<$ une relation d'ordre strict, soit $\leqslant$ l'ordre large associé, $\leqslant$ est totale si et seulement si :
```
I find this easier to read.

File: logique-notes.tex [hunk 3]
```diff
@@ -43,4 +45,146 @@
...
+        \forall x \text{, } y \in E (x < y \text{ ou } y < x \text{ ou } x = y \text{ (Trichotomie)})
+	\end{equation*}
+	\noindent
+
+\end{lemma}
+
+\begin{example}
+    \begin{itemize}
+        \item $x \sqsubseteq y$ si et seulement si $|x| \leqslant |y|$ pour $x$, $y \in \mathbb{C}$, n'est pas anti-symétrique
+            car $-1 \sqsubseteq 1 \sqsubseteq -1$: ce n'est pas une relation d'ordre large.
```
Comment (Yag000) [inline logique-notes.tex]:
```suggestion
            car $-1 \sqsubseteq 1 \sqsubseteq -1$: ce n'est pas une relation d'ordre large car $1 \neq -1$.
```

File: logique-notes.tex [hunk 4]
```diff
@@ -43,4 +45,146 @@
...
+        \item ($\mathscr{P}(\mathbb{N})$, $\subseteq$): Ensemble ordonné d'ordre large pas total (${2} \nsubseteq {17}$ et ${17} \nsubseteq {2}$)
```
Comment (Yag000) [inline logique-notes.tex]:
```suggestion
        \item ($\mathscr{P}(\mathbb{N})$, $\subseteq$): Ensemble ordonné d'ordre large pas total ($\{2\} \nsubseteq \{17\}$ et $\{17\} \nsubseteq \{2\}$)
```

File: logique-notes.tex [hunk 5]
```diff
@@ -43,4 +45,146 @@
...
+
+\subsection{Morphismes d'ordre}
+
+\begin{definition}
+
+    Un morphisme d'ordre entre 2 ensembles d'ordre ($A$, $\leqslant_A$), ($B$, $\leqslant_B$) est une application $\varphi : A \to B$
+    tel que $\forall x$, $y\in A$, $x \leqslant_A y \iff \varphi(x) \leqslant_B \varphi(y)$
+
+\end{definition}
+
```
Comment (Yag000) [inline logique-notes.tex:122]: The definition of isomorphism is missing, I believe

File: logique-notes.tex [hunk 6]
```diff
@@ -43,4 +45,146 @@
...
+        \item Tous sous-ensemble de $E$ non vide a un plus petit élément, c'est-à-dire pour tout $A \subseteq E$, $A \neq \emptyset$, il existe
+            $a \in A$ tel que $\forall b \in A$, $a \leqslant b$
+    \end{itemize}
+
+\end{definition}
+
+\begin{example}
+    \begin{itemize}
+        \item ($\mathbb{N}$, $\leqslant$): Bon ordre
+        \item ($\mathbb{Z}$, $\leqslant$): Mauvais ordre
```
Comment (Yag000) [inline logique-notes.tex]:
```suggestion
        \item ($\mathbb{Z}$, $\leqslant$): Mauvais ordre, car $]-\infty, 0] n'est pas borné à gauche$
```

File: logique-notes.tex [hunk 7]
```diff
@@ -43,4 +45,146 @@
...
+\begin{proof}
+
+    Soit $C \subseteq B$ non vide, alors $C \subseteq A$ non vide. Donc il existe $c \in C$ l'élément le plus petit de $C$ par rapport à $\leqslant_A$. On
+    veut que $c$ soit l'élément le plut petit pour $\leqslant_B$. Soit $d \in C$, donc $c \leqslant_A d$ et puis $c \leqslant_B d$.
+
+\end{proof}
+
+\begin{example}
+    \begin{itemize}
+        \item (${\frac{1}{n} | n \in \mathbb{N}^*}$, $\leqslant$) n'est pas un bon ordre
```
Comment (Yag000) [inline logique-notes.tex]:
```suggestion
        \item ($\left{\frac{1}{n} | n \in \mathbb{N}^*\right}$, $\leqslant$) n'est pas un bon ordre
```

File: logique-notes.tex [hunk 8]
```diff
@@ -43,4 +45,146 @@
...
+        \item (${-\frac{1}{n} | n \in \mathbb{N}^*}$, $\leqslant$) est un bon ordre : il existe un isomorphisme (${-\frac{1}{n} | n \in \mathbb{N}^*}$, $\leqslant$) $\to$ ($\mathbb{N}$, $\leqslant$)
```
Comment (Yag000) [inline logique-notes.tex]:
```suggestion
        \item ($\left{-\frac{1}{n} | n \in \mathbb{N}^*\right}$, $\leqslant$) est un bon ordre : il existe un isomorphisme ($\left{-\frac{1}{n} | n \in \mathbb{N}^*\right}$, $\leqslant$) $\to$ ($\mathbb{N}$, $\leqslant$)
```

File: logique-notes.tex [hunk 9]
```diff
@@ -43,4 +45,146 @@
...
+    \end{equation*}
+	\noindent
+
+\end{prop}
+
+\begin{proof}
+
+    On reformule par contraposée la proposition de récurrence bien fondée :
+    \begin{equation*}
+        \exists x \in E\backslash J \implies \exists y \in E\backslash J \text{ tel que } \forall z \in E \text{, } z < y \implies z \in J \text{ avec } y = min(E\backslash J)
```
Comment (Yag000) [inline logique-notes.tex]:
```suggestion
        \exists x \in E\backslash J \implies \exists y \in E\backslash J \text{ tel que } \forall z \in E \text{, } z < y \implies z \in J \text{ ce qui veut dire que } y = min(E\backslash J)
```

**Review by Yag000 — CHANGES_REQUESTED** [review]
Some modifications needed, but good job nonetheless :)

File: logique-notes.tex (hunk 5 above)
Comment (GabinDDL) [reply logique-notes.tex:122]: yes thanks i forgot it

**Comment by GabinDDL** [comment]
@Yag000 normally it's resolved

File: logique-notes.tex [hunk 10]
```diff
@@ -43,4 +45,157 @@
...
+
+    $<$ est une relation d'ordre strict sur $E$ si elle est :
+    \begin{itemize}
+        \item Irréflexive : $x \nless x$
+        \item Transitive : $\forall x$, $y$, $z \in E$, ($x < y$ et $y < z$) $\implies x < z$
+    \end{itemize}
+    On peut aussi maintenant définir $\leqslant$ de la manière suivante :
+    \begin{itemize}
+        \item $x \leq y$ si et seulement si  $x < y$ ou $x = y$
+    \end{itemize}
```
Comment (Yag000) [inline logique-notes.tex]:
```suggestion
    \begin{equation*}
        x \leq y \iff  x < y \quad \text{ou} \quad  x = y
    \end{equation*}
```

File: logique-notes.tex [hunk 11]
```diff
@@ -43,4 +45,157 @@
...
+    \begin{itemize}
+        \item Réflexive : $\forall x \in E$, $x \leqslant x$
+        \item Anti-symétrique : $\forall x$, $y \in E$, ($x \leqslant y$ et $y \leqslant x$) $\implies x = y$)
+        \item Transitive : $\forall x$, $y$, $z \in E$, ($x \leqslant y$ et $y \leqslant z$) $\implies x \leqslant z$
+    \end{itemize}
+    
+    Aussi, on dit que la relation $<$ sur un ensemble $E$ non vide est une relation d'ordre au sens strict sur $E$ si :
+    \begin{itemize}
+        \item $\forall x$, $y \in E$, $x < y$ si et seulement si $x \leqslant y$ et $x \ne y$
+    \end{itemize}
```
Comment (Yag000) [inline logique-notes.tex]:
```suggestion
    \begin{equation*}
        \forall x,\,y \in E,\,x < y \iff x \leqslant y \quad \text{et} \quad x \ne y
    \end{equation*}
```

File: logique-notes.tex [hunk 12]
```diff
@@ -43,4 +45,157 @@
...
+
+\end{definition}
+
+\begin{lemma}
+
+    Soit $<$ une relation d'ordre strict, soit $\leqslant$ l'ordre large associé, $\leqslant$ est totale si et seulement si :
+	\begin{equation*}
+        \forall x \text{, } y \in E (x < y \text{ ou } y < x \text{ ou } x = y \text{ (Trichotomie)})
+	\end{equation*}
+	\noindent
```
Comment (Yag000) [inline logique-notes.tex]:
```suggestion
```

File: logique-notes.tex [hunk 13]
```diff
@@ -43,4 +45,157 @@
...
+        \item ($\mathbb{Q}$, $\leqslant$): Ensemble ordonné d'ordre large total, et ensemble qui est dense
+        \item ($\mathbb{R}$, $\leqslant$): Ensemble ordonné d'ordre large total, et ensemble qui est dense
+    \end{itemize} 
+\end{example}
+
+\subsection{Morphismes d'ordre}
+
+\begin{definition}
+
+    Un morphisme d'ordre entre 2 ensembles d'ordre ($A$, $\leqslant_A$), ($B$, $\leqslant_B$) est une application $\varphi : A \to B$
```
Comment (Yag000) [inline logique-notes.tex]:
```suggestion
    Un morphisme d'ordres entre 2 ensembles d'ordre ($A$, $\leqslant_A$), ($B$, $\leqslant_B$) est une application $\varphi : A \to B$
```

File: logique-notes.tex [hunk 14]
```diff
@@ -43,4 +45,157 @@
...
+    tel que $\forall x$, $y\in A$, $x \leqslant_A y \iff \varphi(x) \leqslant_B \varphi(y)$
+
+\end{definition}
+
+\begin{definition}
+
+    Un isomorphisme d'ordre est un morphisme d'ordre bijectif.
```
Comment (Yag000) [inline logique-notes.tex]:
```suggestion
    Un isomorphisme d'ordres est un morphisme d'ordres bijectif.
```

File: logique-notes.tex (hunk 12 above)
Comment (Yag000) [reply logique-notes.tex]: Or is there a reason to put it there that I am missing?

**Review by Yag000 — CHANGES_REQUESTED** [review]

File: logique-notes.tex (hunk 12 above)
Comment (GabinDDL) [reply logique-notes.tex]: no you're right, I'll delete it

**Comment by GabinDDL** [comment]
@Yag000 the 2nd part is resolved too

**Review by Yag000 — APPROVED** [review]

**Pull request merged**